beacon/light/api/light_api.go

```diff
@@ -0,0 +1,680 @@
+// https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/beacon-chain.md#syncaggregate
+type syncCommitteeJson struct {
+	Pubkeys   []hexutil.Bytes `json:"pubkeys"`
+	Aggregate hexutil.Bytes   `json:"aggregate_pubkey"`
+}
+
+// serialize returns the serialized version of the committee
+func (s *syncCommitteeJson) serialize() ([]byte, bool) {
+	if len(s.Pubkeys) != params.SyncCommitteeSize {
+		return nil, false
+	}
+	sk := make([]byte, sync.SerializedCommitteeSize)
+	for i, key := range s.Pubkeys {
+		if len(key) != params.BlsPubkeySize {
+			return nil, false
+		}
+		copy(sk[i*params.BlsPubkeySize:(i+1)*params.BlsPubkeySize], key[:])
+	}
+	if len(s.Aggregate) != params.BlsPubkeySize {
+		return nil, false
+	}
+	copy(sk[params.SyncCommitteeSize*params.BlsPubkeySize:], s.Aggregate[:])
+	return sk, true
+}
+
+// GetHead fetches and validates the beacon header with the given blockRoot.
+// If blockRoot is null hash then the latest head header is fetched.
+func (api *BeaconLightApi) GetHeader(blockRoot common.Hash) (types.Header, error) {
+	path := "/eth/v1/beacon/headers/"
+	if blockRoot == (common.Hash{}) {
+		path += "head"
+	} else {
+		path += blockRoot.Hex()
+	}
+	resp, _, err := api.httpGet(path)
+	if err != nil {
+		return types.Header{}, err
+	}
+
+	var data struct {
+		Data struct {
+			Root      common.Hash `json:"root"`
+			Canonical bool        `json:"canonical"`
+			Header    struct {
+				Message   jsonHeader    `json:"message"`
+				Signature hexutil.Bytes `json:"signature"`
+			} `json:"header"`
+		} `json:"data"`
+	}
+	if err := json.Unmarshal(resp, &data); err != nil {
+		return types.Header{}, err
+	}
+	header := data.Data.Header.Message.header()
+	if blockRoot == (common.Hash{}) {
+		blockRoot = data.Data.Root
+	}
+	if header.Hash() != blockRoot {
+		return types.Header{}, errors.New("retrieved beacon header root does not match")
+	}
+	return header, nil
+}
+
+// does not verify state root
+func (api *BeaconLightApi) GetHeadStateProof(format merkle.ProofFormat, paths []string) (merkle.MultiProof, error) {
+	if api.stateProofVersion >= 2 {
+		encFormat, bitLength := EncodeCompactProofFormat(format)
+		return api.getStateProof("head", format, encFormat, bitLength)
+	} else {
+		proof, _, err := api.getOldStateProof("head", format, paths)
+		return proof, err
+	}
+}
+
+type StateProofSub struct {
+	api       *BeaconLightApi
+	format    merkle.ProofFormat
+	paths     []string
+	encFormat []byte
+	bitLength int
+}
+
+func (api *BeaconLightApi) SubscribeStateProof(format merkle.ProofFormat, paths []string, first, period int) (*StateProofSub, error) {
+	if api.stateProofVersion == 0 {
+		return nil, errors.New("State proof API disabled")
+	}
+	encFormat, bitLength := EncodeCompactProofFormat(format)
+	if api.stateProofVersion >= 2 {
+		_, status, err := api.httpGet("/eth/v0/beacon/proof/subscribe/states?format=0x" + hex.EncodeToString(encFormat) + "&first=" + strconv.Itoa(first) + "&period=" + strconv.Itoa(period))
```

Comment:
```suggestion
	_, status, err := api.httpGet("/eth/v0/beacon/proof/subscribe/states?format=#%x&first=%d&period=%d, ",
		encFormat, first, period)
```

... and that suggestion applies to a lot of places in this file. I don't quite get it why you are concatenating strings like that, it's not a common thing to do in go-lang.. ?

Reply:
You could even modify `api.httpGet` to take arguments like `Printf`.. e.g  `httpGetf(format string, a ...any)` and do the interpolation there